src/main.rs

```diff
@@ -129,116 +130,104 @@ fn metadata(path: &Path) -> Result<Metadata, Error> {
+        return Timestamp::new()
+            .store(path.as_path())
+            .context("Failed to write timestamp file");
+    }
 
-        if matches.is_present("stamp") {
-            debug!("Writing timestamp file in: {:?}", path);
-            return Timestamp::new()
-                .store(path.as_path())
-                .context("Failed to write timestamp file");
+    let paths = if args.recursive {
+        find_cargo_projects(&path, args.hidden)
+    } else {
+        let metadata = metadata(&path).context(format!(
+            "Failed to gather metadata for {:?}",
+            path.display()
+        ))?;
+        let out = Path::new(&metadata.target_directory).to_path_buf();
+        if out.exists() {
+            vec![out]
+        } else {
+            anyhow::bail!("Failed to clean {:?} as it does not exist.", out);
         }
+    };
 
-        let paths = if matches.is_present("recursive") {
-            find_cargo_projects(&path, matches.is_present("hidden"))
-        } else {
-            let metadata = metadata(&path).context(format!(
-                "Failed to gather metadata for {:?}",
-                path.display()
-            ))?;
-            let out = Path::new(&metadata.target_directory).to_path_buf();
-            if out.exists() {
-                vec![out]
-            } else {
-                anyhow::bail!("Failed to clean {:?} as it does not exist.", out);
-            }
+    if matches!(criterion, Criterion::Installed | Criterion::Toolchains(_)) {
+        let toolchains = match criterion {
+            Criterion::Installed => vec![],
+            Criterion::Toolchains(vec) => vec,
+            _ => todo!(),
         };
 
-        if matches.is_present("installed") || matches.is_present("toolchains") {
-            for project_path in &paths {
-                match remove_not_built_with(project_path, matches.value_of("toolchains"), dry_run) {
-                    Ok(cleaned_amount) if dry_run => {
-                        info!(
-                            "Would clean: {} from {project_path:?}",
-                            format_bytes(cleaned_amount)
-                        )
-                    }
-                    Ok(cleaned_amount) => info!(
-                        "Cleaned {} from {project_path:?}",
+        for project_path in &paths {
+            match remove_not_built_with(project_path, &toolchains, dry_run) {
+                Ok(cleaned_amount) if dry_run => {
+                    info!(
+                        "Would clean: {} from {project_path:?}",
                         format_bytes(cleaned_amount)
-                    ),
-                    Err(e) => error!(
-                        "{:?}",
-                        e.context(format!("Failed to clean {project_path:?}"))
-                    ),
-                };
-            }
-        } else if matches.is_present("maxsize") {
-            // TODO: consider parsing units like GB, KB ...
```

Comment:
Can you keep this comment in the clap parsing, please?

Reply:
That's about #82, I like replacing TODOs with GitHub issues because I think it gets more visibility and we can also discuss it, what do you think?

Reply:
Issues seem great, but it would be helpful to have a link to the issue here (or a link to the code in the issue).